yjit.rb

```diff
@@ -259,6 +263,19 @@ def add_yjit_hook(hook)
       @yjit_hooks << hook
     end
 
+    # Validates the mem_size and call_threshold options if provided
+    def validate_enable_options(mem_size:, call_threshold:)
+      if mem_size
+        raise ArgumentError, "mem_size must be a Integer" unless mem_size.is_a?(Integer)
+        raise ArgumentError, "mem_size must be between 1 and 2048 MB" unless (1..2048).include?(mem_size)
+      end
+
+      if call_threshold
+        raise ArgumentError, "call_threshold must be a Integer" unless call_threshold.is_a?(Integer)
+        raise ArgumentError, "call_threshold must be a positive integer" unless call_threshold.positive?
+      end
+    end
```

Comment:
nitpick: this doesn't really need to be in a separate method, it just forces people to go look elsewhere to see the validation code